File: backend-golang/internal/model/login_log.go
package model

import "time"

// LoginLog 对应 login_logs 审计表实体。
type LoginLog struct {
	ID         uint64    `gorm:"column:id;primaryKey;autoIncrement"`
	UserID     *uint64   `gorm:"column:user_id"`
	Identifier string    `gorm:"column:identifier;type:varchar(128);not null"`
	Channel    string    `gorm:"column:channel;type:varchar(32);not null"`
	IP         *string   `gorm:"column:ip;type:varchar(45)"`
	UserAgent  *string   `gorm:"column:user_agent;type:varchar(512)"`
	Status     string    `gorm:"column:status;type:varchar(16);not null"`
	CreatedAt  time.Time `gorm:"column:created_at;autoCreateTime"`
}

// TableName 指定 login_logs 表名。
func (LoginLog) TableName() string {
	return "login_logs"
}

File: backend/src/main/java/com/wangning/auth/audit/LoginLog.java
package com.wangning.auth.audit;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * 注册和登录审计日志持久化模型。
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LoginLog {

    /** 日志主键。 */
    private Long id;

    /** 用户 ID，账号不存在时为空。 */
    private Long userId;

    /** 登录或注册使用的标准化账号。 */
    private String identifier;

    /** 登录渠道。 */
    private String channel;

    /** 客户端 IP。 */
    private String ip;

    /** 客户端 User-Agent。 */
    private String userAgent;

    /** 登录结果。 */
    private String status;

    /** 记录时间。 */
    private Instant createdAt;
}
